sdk/core/core-rest-pipeline/test/formDataPolicy.spec.ts

```diff
@@ -0,0 +1,274 @@
+      const result = await performRequest({ a: "va", b: "vb", c: "👻👻" });
+      assert.isUndefined(result.request.formData);
+      const body = result.request.body as any;
+      assert.ok(body, "expecting valid body");
+      assert.ok(isMultipartRequestBody(body), "expecting body to be MultipartRequestBody");
+      const parts = (body as any).parts as BodyPart[];
+      const enc = new TextEncoder();
+      assert.ok(parts.length === 3, "need 3 parts");
+      assert.deepEqual(parts[0], {
+        headers: createHttpHeaders({
+          "Content-Disposition": `form-data; name="a"`,
+        }),
+        body: enc.encode("va"),
+      });
+      assert.deepEqual(parts[1], {
+        headers: createHttpHeaders({
+          "Content-Disposition": `form-data; name="b"`,
+        }),
+        body: enc.encode("vb"),
+      });
+      assert.deepEqual(parts[2], {
+        headers: createHttpHeaders({
+          "Content-Disposition": `form-data; name="c"`,
+        }),
+        body: enc.encode("👻👻"),
+      });
+    });
+
+    describe("file uploads", function () {
+      it("can upload a File object", async function () {
+        if (typeof File === "undefined") {
+          this.skip();
+        }
+
+        const result = await performRequest({
+          file: new File([new Uint8Array([1, 2, 3])], "file.bin", {
+            type: "application/octet-stream",
+          }),
+        });
+
+        const parts = (result.request.body as MultipartRequestBody).parts;
+        assert.ok(parts.length === 1, "expected 1 part");
+        assert.deepEqual(
+          parts[0].headers,
+          createHttpHeaders({
+            "Content-Type": "application/octet-stream",
+            "Content-Disposition": `form-data; name="file"; filename="file.bin"`,
+          })
+        );
+        assert.ok(isBlobLike(parts[0].body));
+        const buf = new Uint8Array(
+          await new Response((parts[0].body as any).stream()).arrayBuffer()
+        );
+        assert.deepEqual([...buf], [1, 2, 3]);
+      });
+
+      it("can upload a Blob object", async function () {
+        if (typeof Blob === "undefined") {
+          this.skip();
+        }
+
+        const result = await performRequest({
+          file: new Blob([new Uint8Array([1, 2, 3])]),
+        });
+
+        const parts = (result.request.body as MultipartRequestBody).parts;
+        assert.ok(parts.length === 1, "expected 1 part");
+        assert.deepEqual(
+          parts[0].headers,
+          createHttpHeaders({
+            // Content-Type should not be inferred
+            "Content-Disposition": `form-data; name="file"; filename="blob"`,
+          })
+        );
+        assert.ok(isBlobLike(parts[0].body));
+        const buf = new Uint8Array(
+          await new Response((parts[0].body as any).stream()).arrayBuffer()
+        );
+        assert.deepEqual([...buf], [1, 2, 3]);
+      });
+
+      it("can upload a Node ReadableStream", async function () {
+        if (!isNode) {
```

Comment:
Could we move this test under a node folder instead so we don't get a bundling warning?